### PR TITLE
feat: sync timestamps and document name.

### PR DESCRIPTION
**Pull request by benoitdevos**
logion-network/logion-internal#191

### CODE REVIEW

File: src/common/Model.ts [hunk 1]
```diff
@@ -116,6 +116,20 @@ export async function fetchLocRequests(
...
 
+export async function fetchLocRequest(
+    axios: AxiosInstance,
+    requestId: string
+): Promise<LocRequest> {
+    const response = await axios.get(`/api/loc-request/${ requestId }`, {
+        headers: {
+            'Cache-Control': 'no-cache',
+            'Pragma': 'no-cache',
+            'Expires': '0',
```
Comment (gdethier) [inline src/common/Model.ts:127]: Sure that the above headers are needed?

File: src/legal-officer/transaction-protection/LocContext.tsx [hunk 2]
```diff
@@ -124,21 +138,105 @@ export interface Props {
...
+    }, [ contextValue.locRequest, contextValue.locId, axios ])
+
+    const enum NextRefresh {
+        STOP,
+        SCHEDULE,
+        IMMEDIATE
+    }
+
+    const refreshNameTimestamp = useCallback<() => Promise<NextRefresh>>(() => {
+        console.log(`refreshNameTimestamp() ${ refreshCounter } ${ new Date() }`)
```
Comment (gdethier) [inline src/legal-officer/transaction-protection/LocContext.tsx:165]: Remove that log?

File: src/legal-officer/transaction-protection/LocContext.tsx [hunk 3]
```diff
@@ -124,21 +138,105 @@ export interface Props {
...
+                    const locFile = findItem(locRequest, locItem)
+                    if (locFile && locFile.addedOn) {
+                        nextRefresh = NextRefresh.IMMEDIATE;
+                        dispatch({ type: 'UPDATE_ITEM_TIMESTAMP', locItem, timestamp: locFile.addedOn })
+                    }
+                })
+            return nextRefresh;
+        }
+        return proceed()
+    }, [ contextValue.loc, contextValue.locItems, contextValue.locId, axios, refreshCounter, NextRefresh.STOP, NextRefresh.SCHEDULE, NextRefresh.IMMEDIATE])
```
Comment (gdethier) [inline src/legal-officer/transaction-protection/LocContext.tsx:215]: I think that moving the enum definition out of the component will enable you to remove the enum items from the dependency array.

File: src/legal-officer/transaction-protection/LocContext.tsx [hunk 4]
```diff
@@ -124,21 +138,105 @@ export interface Props {
...
+                        setTimeout(() => setRefreshing(false), REFRESH_INTERVAL, null)
+                        break;
+                    case NextRefresh.IMMEDIATE:
+                        setRefreshCounter(refreshCounter - 1)
+                        setRefreshing(false)
+                        break;
+                }
+            })()
+        }
+    }, [ refreshNameTimestamp, refreshCounter, setRefreshCounter, refreshing, setRefreshing, NextRefresh.STOP, NextRefresh.SCHEDULE, NextRefresh.IMMEDIATE ])
```
Comment (gdethier) [inline src/legal-officer/transaction-protection/LocContext.tsx:238]: Same remark as above wrt the enum items in the dependency array.

File: src/legal-officer/transaction-protection/LocContext.tsx [hunk 5]
```diff
@@ -124,21 +138,105 @@ export interface Props {
...
 
 export function LocContextProvider(props: Props) {
+
+    const UNKNOWN_FILE = "-";
     const { api } = useLogionChain();
-    const { openedLocRequests, closedLocRequests, axios } = useCommonContext();
+    const { axios } = useCommonContext();
     const [ contextValue, dispatch ] = useReducer(reducer, initialContextValue(props.locId));
+    const [ refreshing, setRefreshing ] = useState<boolean>(false);
+    const [ refreshCounter, setRefreshCounter ] = useState<number>(0);
```
Comment (gdethier) [inline src/legal-officer/transaction-protection/LocContext.tsx:147]: Isn't it possible to include `refreshing` and `refreshCounter` into `contextValue`?

File: src/legal-officer/transaction-protection/LocContext.tsx (hunk 2 above)
Comment (benoitdevos) [reply src/legal-officer/transaction-protection/LocContext.tsx:165]: :+1:

**Pull request merged**